lib/js/jscore.nim

```diff
@@ -110,3 +110,15 @@ proc parse*(l: JsonLib, s: cstring): JsRoot {.importcpp.}
 since (1, 5):
   func debugger*() {.importjs: "debugger@".}
     ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/debugger
+
+  func copyWithin*[T](self: openArray[T]; target: int): seq[T] {.importjs: "#.copyWithin(#)".}
+  func copyWithin*[T](self: openArray[T]; target, start: int): seq[T] {.importjs: "#.copyWithin(#, #)".}
+  func copyWithin*[T](self: openArray[T]; target, start, ends: int): seq[T] {.importjs: "#.copyWithin(#, #, #)".} =
```

Comment:
a `slice: Slice[int]` param would make this easier to use/harder to misuse eg
```nim
func copyWithin*[T](self: openArray[T]; target, slice: Slice[int]): seq[T] {.importjs: "TODO".} =
```

Reply:
Ye, but how to do the importjs pattern, gets complicated.

Reply:
```nim
# only a single importjs proc:
func copyWithinImpl[T](self: openArray[T]; target: int): seq[T] {.importjs: "#.copyWithin(#, #, #)".}
# the rest calls it:
func copyWithin*[T](self: openArray[T]; target: int, slice: Slice[int]): seq[T] =
  copyWithinImpl(self, target, slice.a, slice.b+1)
```

Reply:
But that does not make too much sense, because theres 3 "overloads", with 1, 2, 3 arguments,
that would work for the one with 3, but make the others different and not too consistent. :thinking: